spring-integration-azure/src/test/java/com/microsoft/azure/spring/integration/storage/queue/StorageQueueTemplateReceiveTest.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.spring.integration.storage.queue;
+
+import com.microsoft.azure.spring.integration.core.AzureHeaders;
+import com.microsoft.azure.spring.integration.core.api.CheckpointMode;
+import com.microsoft.azure.spring.integration.storage.queue.factory.StorageQueueClientFactory;
+import com.microsoft.azure.storage.StorageException;
+import com.microsoft.azure.storage.queue.CloudQueue;
+import com.microsoft.azure.storage.queue.CloudQueueMessage;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.messaging.Message;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import static org.junit.Assert.*;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class StorageQueueTemplateReceiveTest {
+
+    @Mock
+    private StorageQueueClientFactory mockClientFactory;
+
+    @Mock
+    private CloudQueue mockClient;
+
+    private StorageQueueOperation operation;
+
+    private CloudQueueMessage cloudQueueMessage = new CloudQueueMessage("test message");
+    private int visibilityTimeoutInSeconds = 30;
+    private String destination = "queue";
+
+    @Before
+    public void setup() {
+
+        when(this.mockClientFactory.getQueueCreator()).thenReturn(t -> this.mockClient);
+        try {
+            when(this.mockClient.retrieveMessage(anyInt(), eq(null), eq(null)))
+                    .thenReturn(this.cloudQueueMessage);
+        } catch (StorageException e) {
+            // StorageException is never thrown here
+        }
+        this.operation = new StorageQueueTemplate(this.mockClientFactory);
+    }
+
+    @Test
+    public void testReceiveFailure() {
+        try {
+            when(this.mockClient.retrieveMessage(eq(visibilityTimeoutInSeconds), eq(null), eq(null)))
+                    .thenThrow(StorageException.class);
+        } catch (StorageException e) {
+            // StorageException is never thrown here
+        }
+
+        CompletableFuture<Message<?>> future =
+                this.operation.receiveAsync(this.destination, this.visibilityTimeoutInSeconds);
+        verifyStorageQueueRuntimeExceptionThrown(future);
+    }
+
+
+    @Test
+    public void testReceiveSuccessWithRecordMode() {
+        CompletableFuture<Message<?>> future = this.operation.receiveAsync(destination);
+        try {
+            assertTrue(Arrays.equals((byte[]) future.get().getPayload(),
+                    this.cloudQueueMessage.getMessageContentAsByte()));
+        } catch (InterruptedException e) {
```

Comment:
I think these exceptions might be handled together. This makes the code much cleaner.